src/hooks/util/useClientReady.ts

```diff
@@ -0,0 +1,9 @@
+import { useState } from "react"
+
+import useClientEffect from "@/hooks/useClientEffect"
+
+export const useClientReady = () => {
+  const [ready, setReady] = useState(false)
```

Comment:
I repeat this a lot to resolve client hydration issues. It's time to abstract it into its own hook.